cosmwasm/utils.js

```diff
@@ -35,20 +35,21 @@ const uploadContract = async (client, wallet, config, options) => {
             return client.upload(account.address, wasm, uploadFee).then(({ checksum, codeId }) => ({ checksum, codeId, account }));
         })
         .then(({ account, checksum, codeId }) => {
+            const usedSalt = salt || contractName.concat(chainNames);
             const address = instantiate2
                 ? instantiate2Address(
                       fromHex(checksum),
                       account.address,
-                      fromHex(getSaltFromKey(salt || contractName.concat(chainNames))),
+                      fromHex(getSaltFromKey(usedSalt)),
```

Comment:
Does the salt need to bytes32, or can it be a string? If the latter, then `getSaltFromKey` isn't needed. We could use the readable string as is